mlrun/artifacts/model.py

```diff
@@ -64,40 +64,54 @@ def __init__(
     ):
 
         super().__init__(key, body, format=format, target_path=target_path)
-        self._inputs: ObjectList = None
-        self._outputs: ObjectList = None
+        self._inputs: typing.Optional[ObjectList] = None
+        self._outputs: typing.Optional[ObjectList] = None
 
         self.model_file = model_file
         self.parameters = parameters or {}
         self.metrics = metrics or {}
-        self.inputs: List[Feature] = inputs or []
-        self.outputs: List[Feature] = outputs or []
+        self.inputs: typing.List[Feature] = inputs or []
+        self.outputs: typing.List[Feature] = outputs or []
         self.extra_data = extra_data or {}
         self.framework = framework
         self.feature_vector = feature_vector
         self.feature_weights = feature_weights
         self.feature_stats = None
 
     @property
-    def inputs(self) -> List[Feature]:
-        """input feature list"""
+    def inputs(self) -> typing.Optional[ObjectList]:
```

Comment:
Why change from `List[Feature]` to `ObjectList` ?

Reply:
Because that is the correct type self._inputs has and thus the property returns. Take a look at the setter's logic, doing `ObjectList.from_list(Feature, inputs)`

Reply:
I wanted to say that this `ObjectList` object should behave exactly like regular python list but I see that's not exactly true 😕 
So I'm not sure what's better to put there 🤷‍♂️

Reply:
better to keep `List[Feature]` so the IDE will give u proper auto complete vs a generic list one

Reply:
@yaronha  I'm not following it's a different type. Putting `List[Feature]` is not what you get on this property. I get why it's better, but that means changing the logic and removing `ObjectList`.